File: flutter_app/lib/models/vehicle.dart
class VehicleModel {
  final String id;
  final String number;
  final String category;
  final int seats;
  final int currentKm;
  final int nextServiceKm;
  final DateTime? fcDate;
  final DateTime? insuranceDate;
  final DateTime? pucDate;

  const VehicleModel({
    required this.id,
    required this.number,
    required this.category,
    required this.seats,
    required this.currentKm,
    required this.nextServiceKm,
    this.fcDate,
    this.insuranceDate,
    this.pucDate,
  });

  factory VehicleModel.fromJson(Map<String, dynamic> json) {
    return VehicleModel(
      id: json['_id'] as String,
      number: json['number'] as String,
      category: (json['category'] as String?) ?? 'sedan',
      seats: (json['seats'] as num?)?.toInt() ?? 4,
      currentKm: (json['currentKm'] as num?)?.toInt() ?? 0,
      nextServiceKm: (json['nextServiceKm'] as num).toInt(),
      fcDate: json['fcDate'] == null ? null : DateTime.tryParse(json['fcDate'].toString()),
      insuranceDate: json['insuranceDate'] == null ? null : DateTime.tryParse(json['insuranceDate'].toString()),
      pucDate: json['pucDate'] == null ? null : DateTime.tryParse(json['pucDate'].toString()),
    );
  }
}

File: backend/models/Vehicle.js
const mongoose = require('mongoose');

const vehicleSchema = new mongoose.Schema(
  {
    number: { type: String, required: true, unique: true, uppercase: true, trim: true },
    category: {
      type: String,
      required: true,
      enum: ['sedan', 'suv', 'mvp', 'van', 'hatchback', 'luxury', 'mini_bus', 'other'],
      default: 'sedan',
    },
    seats: { type: Number, required: true, min: 1, default: 4 },
    fcDate: { type: Date, required: true },
    insuranceDate: { type: Date, required: true },
    pucDate: { type: Date, required: true },
    nextServiceKm: { type: Number, required: true, min: 0 },
    currentKm: { type: Number, default: 0, min: 0 },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Vehicle', vehicleSchema);
